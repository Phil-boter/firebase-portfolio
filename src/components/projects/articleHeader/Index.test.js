import React from "react";
import { shallow, render } from "enzyme";

import ArticleHeader from "./Index.js";

let header = "test-header";
let styling = "test-class";

describe("Articleheader component", () => {
    it("should render Articleheader correctly and match snapshot", () => {
        let wrapper = render(
            <ArticleHeader header={header} styling={styling} />
        );
        expect(wrapper).toMatchSnapshot();
    });
    it("should render H2 with classname='test-class'", () => {
        const wrapper = shallow(
            <ArticleHeader header={header} styling={styling} />
        );
        expect(wrapper.find("h2").hasClass("test-class")).toBe(true);
    });
    it("should render H2 with classname='test-header'.toUpperCase()", () => {
        const wrapper = shallow(
            <ArticleHeader header={header} styling={styling} />
        );
        expect(wrapper.find("h2").hasClass("TEST-HEADER")).toBe(true);
    });
    it("should render H2 with title test-header", () => {
        const wrapper = shallow(
            <ArticleHeader header={header} styling={styling} />
        );
        expect(wrapper.find("h2").text()).toEqual("test-header");
    });
});
