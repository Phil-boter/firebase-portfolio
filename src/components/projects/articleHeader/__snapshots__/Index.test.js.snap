// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Articleheader component should render Articleheader correctly and match snapshot 1`] = `
<h2
  class="test-class TEST-HEADER"
>
  test-header
</h2>
`;
